File: csharp/Discount.cs
namespace supermarket
{

    public class Discount
    {
        private string description;
        private double discountAmount;
        private Product product;

        public Discount(Product product, string description, double discountAmount)
        {
            this.product = product;
            this.description = description;
            this.discountAmount = discountAmount;
        }

        public string getDescription()
        {
            return description;
        }

        public double getDiscountAmount()
        {
            return discountAmount;
        }

        public Product getProduct()
        {
            return product;
        }

    }
}

File: cpp/src/Discount.cpp
#include "Discount.h"
#include "Product.h"

Discount::Discount(const std::string& description, double discountAmount, const Product& product)
        : description(description), discountAmount(discountAmount), product(product) {}

std::string Discount::getDescription() const {
    return description;
}

double Discount::getDiscountAmount() const {
    return discountAmount;
}

Product Discount::getProduct() const {
    return product;
}
